### ISSUE TITLE
bundle fails after installing MySQL 8.0.1 DMR

### ISSUE THREAD

**Issue by yahonda** [brianmario/mysql2#838]
[The MySQL 8.0.1 Milestone Release is available](http://mysqlserverteam.com/the-mysql-8-0-1-milestone-release-is-available/) . I was trying to use it with mysql2 gem. It fails. It used to work witth MySQL 8.0.0 DMR in my environment.

#### Environment:
```
$ more /etc/redhat-release
Fedora release 25 (Twenty Five)
```

* Downloaded https://dev.mysql.com/get/Downloads/MySQL-8.0/mysql-8.0.1-0.1.dmr.fc25.x86_64.rpm-bundle.tar and install MySQL 8.0.1 DMR
```
$ rpm -qa |grep mysql-community
mysql-community-devel-8.0.1-0.1.dmr.fc25.x86_64
mysql-community-client-8.0.1-0.1.dmr.fc25.x86_64
mysql-community-server-8.0.1-0.1.dmr.fc25.x86_64
mysql-community-common-8.0.1-0.1.dmr.fc25.x86_64
mysql-community-test-8.0.1-0.1.dmr.fc25.x86_64
mysql-community-libs-8.0.1-0.1.dmr.fc25.x86_64
$
```

* Ruby and bundler version
```
$ ruby -v
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]
$ bundle -v
Bundler version 1.14.6
```

#### Steps to reproduce:
```
$ git clone https://github.com/rails/rails.git
$ cd rails/activerecord
$ bundle
```
#### Results:
```
$ bundle
Fetching gem metadata from https://rubygems.org/.........
Fetching version metadata from https://rubygems.org/..
Fetching dependency metadata from https://rubygems.org/.
Using rake 12.0.0
Using concurrent-ruby 1.0.4
Using i18n 0.8.1
Using minitest 5.3.3
Using thread_safe 0.3.5
Using builder 3.2.3
Using erubi 1.6.0
Using mini_portile2 2.1.0
Using rack 2.0.1
Using nio4r 2.0.0
Using websocket-extensions 0.1.2
Using mime-types-data 3.2016.0521
Using arel 8.0.0 from https://github.com/rails/arel.git (at master@437aa3a)
Using public_suffix 2.0.5
Using amq-protocol 2.1.0
Using ast 2.3.0
Using beaneater 1.0.0
Using dante 0.2.0
Using bcrypt 3.1.11
Using benchmark-ips 2.7.2
Using blade-qunit_adapter 2.0.1
Using coffee-script-source 1.12.2
Using execjs 2.7.0
Using curses 1.0.2
Using eventmachine 1.2.1
Using cookiejar 0.3.3
Using http_parser.rb 0.6.0
Using multi_json 1.12.1
Using daemons 1.2.4
Using thor 0.19.4
Using useragent 0.16.8
Using ffi 1.9.17
Using multipart-post 2.0.0
Using rubyzip 1.2.0
Using websocket 1.2.3
Using byebug 9.0.6
Using method_source 0.8.2
Using connection_pool 2.2.1
Using dalli 2.7.6
Using hiredis 0.6.1
Using erubis 2.7.0
Using event_emitter 0.2.5
Using json 2.0.3
Using mustache 1.0.3
Using libxml-ruby 2.9.0
Using rb-fsevent 0.9.8
Using ruby_dep 1.5.0
Using metaclass 0.0.4
Using mono_logger 1.1.0
Installing mysql2 0.4.5 with native extensions
Using pg 0.19.0
Using powerpack 0.1.1
Using puma 3.7.0
Using redis 3.3.2
Using simple_uuid 0.4.0
Using que 0.12.0
Using racc 1.4.14
Using bundler 1.14.6
Using rainbow 2.2.1
Using rdoc 5.0.0
Using redcarpet 3.2.3
Using ruby-progressbar 1.8.1
Using unicode-display_width 1.1.3
Using sass 3.4.23
Using tilt 2.0.5
Using sequel 4.42.1
Using sigdump 0.2.4
Using thread 0.1.7
Using sprockets-export 0.9.1
Using sqlite3 1.3.13
Using stackprof 0.2.10
Using turbolinks-source 5.0.0
Using sucker_punch 2.0.2
Using tzinfo 1.2.2
Using nokogiri 1.7.0.1
Using rack-test 0.6.3
Using sprockets 3.7.1
Using rack-cache 1.6.1
Using rack-protection 1.5.3
Using sinatra 1.0
Using vegas 0.1.11
Using websocket-driver 0.6.4
Using mime-types 3.1
Using addressable 2.5.0
Using bunny 2.6.2
Using parser 2.4.0.0
Using backburner 1.3.1
Using coffee-script 2.4.1
Using uglifier 3.0.4
Using em-socksify 0.3.1
Using qu 0.2.0
Using thin 1.7.0
Using childprocess 0.5.9
Using rb-inotify 0.9.7 from https://github.com/matthewd/rb-inotify.git (at close-handling@9055351)
Using faraday 0.11.0
Using em-hiredis 0.3.1
Using websocket-client-simple 0.3.0 from https://github.com/matthewd/websocket-client-simple.git (at close-race@e161305)
Using mocha 0.14.0
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/mysql2-0.4.5/ext/mysql2
/home/yahonda/.rbenv/versions/2.4.1/bin/ruby -r ./siteconf20170411-17346-uwyi9q.rb extconf.rb
checking for rb_absint_size()... yes
checking for rb_absint_singlebit_p()... yes
checking for ruby/thread.h... yes
checking for rb_thread_call_without_gvl() in ruby/thread.h... yes
checking for rb_thread_blocking_region()... no
checking for rb_wait_for_single_fd()... yes
checking for rb_hash_dup()... yes
checking for rb_intern3()... yes
checking for rb_big_cmp()... yes
-----
Using mysql_config at /usr/bin/mysql_config
-----
checking for mysql.h... yes
checking for SSL_MODE_DISABLED in mysql.h... yes
checking for SSL_MODE_PREFERRED in mysql.h... yes
checking for SSL_MODE_REQUIRED in mysql.h... yes
checking for SSL_MODE_VERIFY_CA in mysql.h... yes
checking for SSL_MODE_VERIFY_IDENTITY in mysql.h... yes
checking for errmsg.h... yes
checking for mysqld_error.h... yes
-----
Setting libpath to /usr/lib64/mysql
-----
creating Makefile

current directory: /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/mysql2-0.4.5/ext/mysql2
make "DESTDIR=" clean

current directory: /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/mysql2-0.4.5/ext/mysql2
make "DESTDIR="
compiling client.c
In file included from ./mysql2_ext.h:41:0,
                 from client.c:1:
./result.h:24:3: error: unknown type name ‘my_bool’
   my_bool *is_null;
   ^~~~~~~
./result.h:25:3: error: unknown type name ‘my_bool’
   my_bool *error;
   ^~~~~~~
client.c: In function ‘nogvl_read_query_result’:
client.c:507:3: error: unknown type name ‘my_bool’
   my_bool res = mysql_read_query_result(client);
   ^~~~~~~
client.c: In function ‘_mysql_client_options’:
client.c:828:3: error: unknown type name ‘my_bool’
   my_bool boolval;
   ^~~~~~~
client.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-used-but-marked-unused’
cc1: warning: unrecognized command line option ‘-Wno-static-in-inline’
cc1: warning: unrecognized command line option ‘-Wno-reserved-id-macro’
cc1: warning: unrecognized command line option ‘-Wno-missing-variable-declarations’
cc1: warning: unrecognized command line option ‘-Wno-documentation-unknown-command’
cc1: warning: unrecognized command line option ‘-Wno-disabled-macro-expansion’
cc1: warning: unrecognized command line option ‘-Wno-covered-switch-default’
cc1: warning: unrecognized command line option ‘-Wno-conditional-uninitialized’
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
Makefile:241: recipe for target 'client.o' failed
make: *** [client.o] Error 1

make failed, exit code 2

Gem files will remain installed in /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/mysql2-0.4.5 for inspection.
Results logged to /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0-static/mysql2-0.4.5/gem_make.out

An error occurred while installing mysql2 (0.4.5), and Bundler cannot continue.
Make sure that `gem install mysql2 -v '0.4.5'` succeeds before bundling.
$
```

* gem_make.out
```
$ more /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0-static/mysql2-0.4.5/gem_make.out
current directory: /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/mysql2-0.4.5/ext/mysql2
/home/yahonda/.rbenv/versions/2.4.1/bin/ruby -r ./siteconf20170411-17346-uwyi9q.rb extconf.rb
checking for rb_absint_size()... yes
checking for rb_absint_singlebit_p()... yes
checking for ruby/thread.h... yes
checking for rb_thread_call_without_gvl() in ruby/thread.h... yes
checking for rb_thread_blocking_region()... no
checking for rb_wait_for_single_fd()... yes
checking for rb_hash_dup()... yes
checking for rb_intern3()... yes
checking for rb_big_cmp()... yes
-----
Using mysql_config at /usr/bin/mysql_config
-----
checking for mysql.h... yes
checking for SSL_MODE_DISABLED in mysql.h... yes
checking for SSL_MODE_PREFERRED in mysql.h... yes
checking for SSL_MODE_REQUIRED in mysql.h... yes
checking for SSL_MODE_VERIFY_CA in mysql.h... yes
checking for SSL_MODE_VERIFY_IDENTITY in mysql.h... yes
checking for errmsg.h... yes
checking for mysqld_error.h... yes
-----
Setting libpath to /usr/lib64/mysql
-----
creating Makefile

current directory: /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/mysql2-0.4.5/ext/mysql2
make "DESTDIR=" clean

current directory: /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/mysql2-0.4.5/ext/mysql2
make "DESTDIR="
compiling client.c
In file included from ./mysql2_ext.h:41:0,
                 from client.c:1:
./result.h:24:3: error: unknown type name ‘my_bool’
   my_bool *is_null;
   ^~~~~~~
./result.h:25:3: error: unknown type name ‘my_bool’
   my_bool *error;
   ^~~~~~~
client.c: In function ‘nogvl_read_query_result’:
client.c:507:3: error: unknown type name ‘my_bool’
   my_bool res = mysql_read_query_result(client);
   ^~~~~~~
client.c: In function ‘_mysql_client_options’:
client.c:828:3: error: unknown type name ‘my_bool’
   my_bool boolval;
   ^~~~~~~
client.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-used-but-marked-unused’
cc1: warning: unrecognized command line option ‘-Wno-static-in-inline’
cc1: warning: unrecognized command line option ‘-Wno-reserved-id-macro’
cc1: warning: unrecognized command line option ‘-Wno-missing-variable-declarations’
cc1: warning: unrecognized command line option ‘-Wno-documentation-unknown-command’
cc1: warning: unrecognized command line option ‘-Wno-disabled-macro-expansion’
cc1: warning: unrecognized command line option ‘-Wno-covered-switch-default’
cc1: warning: unrecognized command line option ‘-Wno-conditional-uninitialized’
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
Makefile:241: recipe for target 'client.o' failed
make: *** [client.o] Error 1

make failed, exit code 2
$
```
</details>

**Comment by yahonda** [comment]
Looks like `my_bool` has been renamed to `bool` in MySQL 8.0. 

https://dev.mysql.com/doc/refman/8.0/en/c-api-prepared-statement-data-structures.html

Renamed `my_bool` to `bool`  https://github.com/yahonda/mysql2/commit/a2df25be57b3f9cc07ab71f1080f3866b8b93f88

and it looks fine except for SSL one. 

```ruby
$ bundle exec rake spec
install -c tmp/x86_64-linux/mysql2/2.4.1/mysql2.so lib/mysql2/mysql2.so
cp tmp/x86_64-linux/mysql2/2.4.1/mysql2.so tmp/x86_64-linux/stage/lib/mysql2/mysql2.so
/home/yahonda/.rbenv/versions/2.4.1/bin/ruby -I/home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rspec-core-3.5.4/lib:/home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rspec-support-3.5.0/lib /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rspec-core-3.5.4/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb

Randomized with seed 62812

Mysql2::Statement
  should keep its result after other query
  should handle bignum but beyond int64_t
  should be reusable 10000 times
  should raise an exception when server disconnects
  should tell us the result count
  should tell us the field count
  should prepare Date values
  should tell us the param count
  should create a statement
  should raise an exception without a block
  should be reusable 1000 times
  should update a DECIMAL value passing a BigDecimal
  should select dates
  should handle bignum but in int64_t
  should let us iterate over results
  should tell us about the fields
  should handle comparisons and likes
  should let us execute our statement
  should prepare DateTime values with microseconds
  should handle as a decimal binding a BigDecimal
  should prepare Time values with microseconds
  should warn but still work if cache_rows is set to false
  #each
    should throw an exception if we try to iterate twice when streaming is enabled
    should yield rows as hash's with symbol keys if :symbolize_keys was set to true
    should be able to return results as an array
    should yield different value for #first if streaming
    should yield rows as hash's
    should yield the same value for #first if streaming is disabled
    should cache previously yielded results by default
  last_id
    should return last insert id
    should handle bigint ids
  affected_rows
    should return number of rows affected by an insert
    should return number of rows affected by a delete
    should return number of rows affected by an update
  #fields
    method should exist
    should return an array of field names in proper order
  close
    should raise an error on subsequent execution
    should free server resources
  row data type mapping
    should return TrueClass or FalseClass for a TINYINT value if :cast_booleans is enabled
    should return a String for VARCHAR
    should return a String for TEXT
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:418: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:418: warning: constant ::Bignum is deprecated
    should return Fixnum for a BIGINT value
    should return Time for a TIME value
    should return TrueClass or FalseClass for a BIT(1) value if :cast_booleans is enabled
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:423: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:423: warning: constant ::Bignum is deprecated
    should return Fixnum for a YEAR value
    should return a String for BLOB
    should return a String for VARBINARY
    should return String for an ENUM value
    should return Time for a TIMESTAMP value when within the supported range
    should return Time when timestamp is > 2038-01-19T03:14:07
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:408: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:408: warning: constant ::Bignum is deprecated
    should return Fixnum for a MEDIUMINT value
    should return Time for a DATETIME value when within the supported range
    should return nil for a NULL value
    should return String for a SET value
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:413: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:413: warning: constant ::Bignum is deprecated
    should return Fixnum for an INT value
    should return Date for a DATE value
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:403: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:403: warning: constant ::Bignum is deprecated
    should return Fixnum for a SMALLINT value
    should return a String for LONGBLOB
    should return a String for MEDIUMBLOB
    should return a String for TINYBLOB
    should return Float for a DOUBLE value
    should return String for a BIT(64) value
    should return String for a BINARY value
    should return BigDecimal for a DECIMAL value
    should return Time when timestamp is < 1901-12-13 20:45:52
    should raise an error given an invalid DATETIME
    should return a String for LONGTEXT
    should return a String for TINYTEXT
    should return Float for a FLOAT value
    should return a String for CHAR
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:366: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/statement_spec.rb:366: warning: constant ::Bignum is deprecated
    should return Fixnum for a TINYINT value
    should return String for a BIT(1) value
    should return a String for MEDIUMTEXT
    string encoding for BINARY values
      should default to binary if Encoding.default_internal is nil
      should not use Encoding.default_internal
    string encoding for VARBINARY values
      should not use Encoding.default_internal
      should default to binary if Encoding.default_internal is nil
    string encoding for MEDIUMTEXT values
      should use Encoding.default_internal
      should default to utf-8 if Encoding.default_internal is nil
    string encoding for SET values
      should use Encoding.default_internal
      should default to the connection's encoding if Encoding.default_internal is nil
    string encoding for TEXT values
      should default to utf-8 if Encoding.default_internal is nil
      should use Encoding.default_internal
    string encoding for ENUM values
      should use Encoding.default_internal
      should default to the connection's encoding if Encoding.default_internal is nil
    string encoding for MEDIUMBLOB values
      should default to binary if Encoding.default_internal is nil
      should not use Encoding.default_internal
    string encoding for TINYTEXT values
      should default to utf-8 if Encoding.default_internal is nil
      should use Encoding.default_internal
    string encoding for LONGBLOB values
      should default to binary if Encoding.default_internal is nil
      should not use Encoding.default_internal
    string encoding for LONGTEXT values
      should default to utf-8 if Encoding.default_internal is nil
      should use Encoding.default_internal
    string encoding for BLOB values
      should default to binary if Encoding.default_internal is nil
      should not use Encoding.default_internal
    string encoding for CHAR values
      should default to utf-8 if Encoding.default_internal is nil
      should use Encoding.default_internal
    string encoding for VARCHAR values
      should use Encoding.default_internal
      should default to utf-8 if Encoding.default_internal is nil
    string encoding for TINYBLOB values
      should not use Encoding.default_internal
      should default to binary if Encoding.default_internal is nil
  utf8_db
    should be able to retrieve utf8 param query correctly
    should be able to retrieve query with param in different encoding correctly
    should be able to retrieve utf8 field names correctly
  streaming result
    should be able to stream query result

Mysql2::Error
  responds to error_number and sql_state, with aliases
  encoding
    returns error messages as UTF-8 by default
    returns sql state as ASCII
    returns error messages and sql state in Encoding.default_internal if set

Mysql2::Result
  should raise a Mysql2::Error exception upon a bad query
  should respond to #each
  should raise a TypeError exception when it doesn't wrap a result set
  should be able to return the number of rows in the result set
  should respond to #free
  should have included Enumerable
  should respond to #count, which is aliased as #size
  #each
    should yield rows as hash's with symbol keys if :symbolize_keys was set to true
    should cache previously yielded results by default
    should yield different value for #first if streaming
    should be able to return results as an array
    should not cache previously yielded results if cache_rows is disabled
    should throw an exception if we try to iterate twice when streaming is enabled
    should be able to iterate a second time even if cache_rows is disabled
    should yield rows as hash's
    should yield the same value for #first if streaming is disabled
  streaming
    #count should be zero for rows after streaming when there were no results
    should raise an exception if streaming ended due to a timeout
    should retain the count when mixing first and each
    should maintain a count while streaming
    should not yield nil at the end of streaming
  metadata queries
    should show tables
  row data type mapping
    should return Time when timestamp is < 1901-12-13 20:45:52
    should return a String for TINYTEXT
    should return TrueClass or FalseClass for a TINYINT value if :cast_booleans is enabled
    should return Time when timestamp is > 2038-01-19T03:14:07
    should return String for a BIT(1) value
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:209: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:209: warning: constant ::Bignum is deprecated
    should return Fixnum for a TINYINT value
    should return a String for MEDIUMTEXT
    should return a String for LONGTEXT
    should return a String for MEDIUMBLOB
    should return Time for a TIMESTAMP value when within the supported range
    should return a String for CHAR
    should return TrueClass or FalseClass for a BIT(1) value if :cast_booleans is enabled
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:266: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:266: warning: constant ::Bignum is deprecated
    should return Fixnum for a YEAR value
    should return a String for TEXT
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:261: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:261: warning: constant ::Bignum is deprecated
    should return Fixnum for a BIGINT value
    should return Float for a DOUBLE value
    should raise an error given an invalid DATETIME
    should return Date for a DATE value
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:256: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:256: warning: constant ::Bignum is deprecated
    should return Fixnum for an INT value
    should return a String for TINYBLOB
    should return a String for VARCHAR
    should return String for a BINARY value
    should return a String for BLOB
    should return a String for LONGBLOB
    should return String for a BIT(64) value
    should return Float for a FLOAT value
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:246: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:246: warning: constant ::Bignum is deprecated
    should return Fixnum for a SMALLINT value
    should return nil for a NULL value
    should return a String for VARBINARY
    should return String for an ENUM value
    should return Time for a DATETIME value when within the supported range
    should return String for a SET value
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:251: warning: constant ::Fixnum is deprecated
/home/yahonda/git/mysql2/spec/mysql2/result_spec.rb:251: warning: constant ::Bignum is deprecated
    should return Fixnum for a MEDIUMINT value
    should return BigDecimal for a DECIMAL value
    should return Time for a TIME value
    should return nil values for NULL and strings for everything else when :cast is false
    string encoding for CHAR values
      should use Encoding.default_internal
      should default to utf-8 if Encoding.default_internal is nil
    string encoding for MEDIUMBLOB values
      should not use Encoding.default_internal
      should default to binary if Encoding.default_internal is nil
    string encoding for BLOB values
      should default to binary if Encoding.default_internal is nil
      should not use Encoding.default_internal
    string encoding for TEXT values
      should use Encoding.default_internal
      should default to utf-8 if Encoding.default_internal is nil
    string encoding for TINYTEXT values
      should default to utf-8 if Encoding.default_internal is nil
      should use Encoding.default_internal
    string encoding for BINARY values
      should not use Encoding.default_internal
      should default to binary if Encoding.default_internal is nil
    string encoding for SET values
      should use Encoding.default_internal
      should default to the connection's encoding if Encoding.default_internal is nil
    string encoding for ENUM values
      should default to the connection's encoding if Encoding.default_internal is nil
      should use Encoding.default_internal
    string encoding for TINYBLOB values
      should not use Encoding.default_internal
      should default to binary if Encoding.default_internal is nil
    string encoding for LONGBLOB values
      should not use Encoding.default_internal
      should default to binary if Encoding.default_internal is nil
    string encoding for LONGTEXT values
      should use Encoding.default_internal
      should default to utf-8 if Encoding.default_internal is nil
    string encoding for VARCHAR values
      should use Encoding.default_internal
      should default to utf-8 if Encoding.default_internal is nil
    string encoding for VARBINARY values
      should default to binary if Encoding.default_internal is nil
      should not use Encoding.default_internal
    string encoding for MEDIUMTEXT values
      should use Encoding.default_internal
      should default to utf-8 if Encoding.default_internal is nil
  #fields
    should return an array of field names in proper order
    method should exist

Mysql2::Client
  should parse flags string
  should allow nil read_timeout
  should expect write_timeout to be a positive integer
  should parse flags array
  should respond to #query_info
  should raise an exception on create for invalid encodings
  #thread_id should return a boolean
  #server_info should require an open connection
  should be able to connect to database with numeric-only name
  should default flags to (REMEMBER_OPTIONS, LONG_PASSWORD, LONG_FLAG, TRANSACTIONS, PROTOCOL_41, SECURE_CONNECTION)
  should respond to #socket
  should raise an exception on non-string encodings
  should expect read_timeout to be a positive integer
  should respond to #escape
  should respond to #warning_count
  should be able to connect via SSL options (FAILED - 1)
  should expect connect_timeout to be a positive integer
  should execute init command
/home/yahonda/git/mysql2/spec/mysql2/client_spec.rb:817: warning: constant ::Fixnum is deprecated
  #info should return a hash containing the client version ID and String
  should terminate connections when calling close
/home/yahonda/git/mysql2/spec/mysql2/client_spec.rb:939: warning: constant ::Fixnum is deprecated
  #thread_id should be a Fixnum
  should respond to #close
  should accept connect flags and pass them to #connect
  should respond to #ping
  should not raise an exception on create for a valid encoding
  should be able to close properly
/home/yahonda/git/mysql2/spec/mysql2/client_spec.rb:848: warning: constant ::Fixnum is deprecated
  #server_info should return a hash containing the client version ID and String
  should have a global default_query_options hash
  should respond to #query
  should send init_command after reconnect
  should respond to #thread_id
  should respond to #info
  should respond to escape
  should raise an exception upon connection failure
  should respond to #encoding
  should raise a Mysql2::Error exception upon connection failure
  should not leave dangling connections after garbage collection
  should respond to #server_info
  escape
    should not overflow the thread stack
    should return the passed string if nothing was escaped
    should not overflow the process stack
    should carry over the original string's encoding
    should return a new SQL-escape version of the passed string
  #warning_count
    when no warnings
      should 0
    when has a warnings
      should > 0
  #escape
    should not overflow the thread stack
    should return the passed string if nothing was escaped
    should require an open connection
    should return a new SQL-escape version of the passed string
    should not overflow the process stack
    when mysql encoding is not utf8
      should return a internal encoding string if Encoding.default_internal is set
  strings returned by #info
    should be tagged as ascii
  strings returned by .info
    should be tagged as ascii
  #query_info
    when no info present
      should 0
    when has some info
      should retrieve it
  #query
    should be able to return results as an array
    should be impervious to connection-corrupting timeouts in #query
/home/yahonda/git/mysql2/spec/mysql2/client_spec.rb:530: warning: constant ::Fixnum is deprecated
    #socket should return a Fixnum (file descriptor from C)
    #socket should require an open connection
    threaded queries should be supported
    should only accept strings as the query parameter
    should let you query again if iterating is finished when streaming
    should describe the thread holding the active query
    should not allow another query to be sent without fetching a result first
    evented async queries should be supported
    should not retain query options set on a query for subsequent queries, but should retain it in the result
    should not let you query again if iterating is not finished when streaming
    should allow changing query options for subsequent queries
    should return results as a hash by default
    should require an open connection
    should timeout if we wait longer than :read_timeout
    should be able to return results with symbolized keys
    should run signal handlers while waiting for a response
    should be impervious to connection-corrupting timeouts in #execute
    when a non-standard exception class is raised
      should close the connection when an exception is raised
      should handle Timeouts without leaving the connection hanging if reconnect is set to true after construction
      should handle Timeouts without leaving the connection hanging if reconnect is true
    Multiple results sets
      returns multiple result sets
      #abandon_results! should work
      should raise an exception when one of multiple statements fails
      #more_results? should work with stored procedures
      will raise on query if there are outstanding results to read
      #more_results? should work
      does not interfere with other statements
  select_db
    should raise a Mysql2::Error when the database doesn't exist
    should switch databases
    should respond to #select_db
    should return the database switched to
  strings returned by #server_info
    should use Encoding.default_internal
    should default to the connection's encoding if Encoding.default_internal is nil
  #automatic_close
    is enabled by default
    can be configured
    should not close connections when running in a child process
    can be assigned
  using defaults file
    should not raise an exception for valid defaults group
    should not raise an exception without default group
  write operations api
    should respond to #last_id
    #last_id should return a Fixnum, the from the last INSERT/UPDATE
    should respond to #last_id
    #last_id should return a Fixnum, the from the last INSERT/UPDATE
    #last_id should handle BIGINT auto-increment ids above 32 bits
  :local_infile
    should LOAD DATA LOCAL INFILE
    should raise an error when a non-existent file is loaded
    should raise an error when local_infile is disabled

Mysql2::EM::Client
  should not raise error when closing client with no query running
  should not swallow exceptions raised in callbacks
  should support queries in callbacks
  should support async queries
  when an exception is raised by the client
    should fail the deferrable
    should swallow exceptions raised in by the client

Failures:

  1) Mysql2::Client should be able to connect via SSL options
     Failure/Error:
       expect {
         # rubocop:disable Style/TrailingComma
         ssl_client = Mysql2::Client.new(
           DatabaseCredentials['root'].merge(
             'host'     => 'mysql2gem.example.com', # must match the certificates
             :sslkey    => '/etc/mysql/client-key.pem',
             :sslcert   => '/etc/mysql/client-cert.pem',
             :sslca     => '/etc/mysql/ca-cert.pem',
             :sslcipher => 'DHE-RSA-AES256-SHA',
             :sslverify => true

       expected no Exception, got #<Mysql2::Error: Unknown MySQL server host 'mysql2gem.example.com' (95)> with backtrace:
         # ./lib/mysql2/client.rb:89:in `connect'
         # ./lib/mysql2/client.rb:89:in `initialize'
         # ./spec/mysql2/client_spec.rb:141:in `new'
         # ./spec/mysql2/client_spec.rb:141:in `block (3 levels) in <top (required)>'
         # ./spec/mysql2/client_spec.rb:139:in `block (2 levels) in <top (required)>'
     # ./spec/mysql2/client_spec.rb:139:in `block (2 levels) in <top (required)>'

Finished in 10.1 seconds (files took 0.2643 seconds to load)
307 examples, 1 failure

Failed examples:

rspec ./spec/mysql2/client_spec.rb:130 # Mysql2::Client should be able to connect via SSL options

Randomized with seed 62812

/home/yahonda/.rbenv/versions/2.4.1/bin/ruby -I/home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rspec-core-3.5.4/lib:/home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rspec-support-3.5.0/lib /home/yahonda/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rspec-core-3.5.4/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb failed
$
```

**Comment by yahonda** [comment]
Closed as #840 has been merged.
